utils/src/main/kotlin/com/exactpro/th2/lwdataprovider/MessageSearcher.kt

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.lwdataprovider
+
+import com.exactpro.th2.common.grpc.Direction
+import com.exactpro.th2.common.message.toJson
+import com.exactpro.th2.common.message.toTimestamp
+import com.exactpro.th2.dataprovider.lw.grpc.DataProviderService
+import com.exactpro.th2.dataprovider.lw.grpc.MessageGroupsSearchRequest
+import com.exactpro.th2.dataprovider.lw.grpc.MessageSearchResponse
+import com.exactpro.th2.dataprovider.lw.grpc.MessageStream
+import com.exactpro.th2.dataprovider.lw.grpc.TimeRelation
+import io.grpc.Context
+import mu.KotlinLogging
+import java.time.Duration
+import java.time.Instant
+import java.util.function.Function
+
+/**
+ * This class provide ability to search messages via lw-data-provider gRPC service
+ * @param requestInterval interval for sending requests via [service]. Searcher uses this value to split search interval to smaller pieces to reduce volume of requested data via one time.
+ */
+class MessageSearcher @JvmOverloads constructor(
```

Comment:
I would prefer if we make the constructor private and create factory method(s) to create the searcher. This will give us more control on what should be passed into the MessageSeacher constructor and should help with any migrations in the future